parser/meta.go

```diff
@@ -573,3 +589,48 @@ func sortedRefs(refs map[ast.Node]*est.Node) []refPair {
 	})
 	return p
 }
+
+type selectorMap struct {
+	services map[selector.Selector]map[*est.Service]struct{}
+	rpcs     map[selector.Selector]map[*est.RPC]struct{}
+}
+
+func newSelectorMap() *selectorMap {
```

Comment:
We have something similar to this in `parser/est/selectors.go` but in the opposite direction: given an RPC, which middleware(s) apply to it. Should we move this piece of code there and generalize it to support both use cases?

Reply:
I don't think so for now, one is a generalised lookup cache, the other is a one time search